src/vanna/weaviate/weaviate_vector.py

```diff
@@ -7,7 +7,7 @@
 
 class WeaviateDatabase(VannaBase):
 
-    def __init__(self, config=None):
+    def __init__(self, config=None,client= None):
```

Comment:
If you want to set the client directly, you should aim to do something like that is done for Chroma.

See here:
https://github.com/vanna-ai/vanna/blob/main/src/vanna/chromadb/chromadb_vector.py#L23

and here:
https://github.com/vanna-ai/vanna/blob/main/src/vanna/chromadb/chromadb_vector.py#L38

So basically, pass the client as part of the config, and then you likely want to update what is done here, if the client is part of the config:
https://github.com/vanna-ai/vanna/pull/654/files#diff-acb7ec3a110767dae2ec2a321b040ea390d7165a898d49c6ce310180ece936daR39

See the chroma client implementation for inspiration.